src/qt/transactionrecord.cpp

```diff
@@ -53,30 +53,56 @@ QList<TransactionRecord> TransactionRecord::decomposeTransaction(const CWallet *
     }
 
     if (wtx.IsZerocoinSpend() || isAllSigmaSpendFromMe) {
+        CAmount nTxFee = nDebit - wtx.GetValueOut();
+        bool first = true;
+
+        bool isAllToMe = true;
         for (const CTxOut& txout : wtx.vout) {
-            if (wtx.IsChange(txout)) {
-                continue;
+            if (!wallet->IsMine(txout)) {
+                isAllToMe = false;
+                break;
             }
-            isminetype mine = wallet->IsMine(txout);
+        }
 
+        if(isAllToMe){
             TransactionRecord sub(hash, nTime);
             CTxDestination address;
+            sub.involvesWatchAddress = ISMINE_WATCH_ONLY;
+            if (ExtractDestination(wtx.vout[0].scriptPubKey, address) && IsMine(*wallet, address)) {
+                sub.type = TransactionRecord::SpendToSelf;
+                sub.address = CBitcoinAddress(address).ToString();
```

Comment:
I think we should not show only the first address. My idea is:

1. If there is one output then just show it address.
2. If there is more than one outputs. We have two choices here:
    1. Show all addresses separated by some symbol (e.g. `,`). For me this is the best choice.
    2. Just show something like `Multiple addresses`.

Reply:
Agree